data-management/data-interface.js

```diff
@@ -249,19 +258,12 @@ const editUser = async (parameters, context) => {
 
 const updateMyUser = async (parameters, context) => {
     formatParams(parameters);
-    try {
-        let userInfo = context.userInfo;
-        if (!userInfo) {
-            return new Error(errorName.NOT_LOGGED_IN);
-        }
-        else{
-            parameters = {...userInfo, ...parameters.userInfo};
-            parameters.editDate = (new Date()).toString()
-            return await neo4j.updateMyUser(parameters, userInfo);
-        }
-    } catch (err) {
-        return err;
-    }
+    const task = async () => {
+        validator.isValidLoginOrThrow(context.userInfo);
+        const user = UserBuilder.createUser(context.userInfo);

```

Comment:
In this case, creating a user object seems unnecessary. We are passing in context.userInfo for initialization and then retrieving the same information from the user in neo4j-service.js . Instead we could skip this and just pass the context.userInfo object as the user parameter and retrieve the admin email and IDP directly from that.